src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -2243,15 +2242,16 @@ private bool getReadonlyStaticFieldValue(CORINFO_FIELD_STRUCT_* fieldHandle, byt
         private CORINFO_CLASS_STRUCT_* getObjectType(void* objPtr)
         {
             object obj = HandleToObject((IntPtr)objPtr);
-            if (obj is FrozenStringNode)
+            return obj switch
             {
-                return ObjectToHandle(_compilation.TypeSystemContext.GetWellKnownType(WellKnownType.String));
-            }
-            else if (obj is FrozenObjectNode)
-            {
-                return null; // TODO: how to implement properly?
-            }
-            throw new NotImplementedException($"Unexpected object in getObjectType: {obj}");
+                // For now we only support frozen strings
+                FrozenStringNode => ObjectToHandle(_compilation.TypeSystemContext.GetWellKnownType(WellKnownType.String)),
+
+                // and frozen objects like arrays
+                FrozenObjectNode frozenObj => ObjectToHandle(frozenObj.OwningType),
```

Comment:
That's the type that owns the data blob, not the type of the field

Reply:
Ah, I assume I need to expose `_data.Type` instead?